### PR TITLE
Convert to latest Swift

### PR DESCRIPTION
**Pull request by rshevchuk**

### CODE REVIEW

File: DominantColor/Shared/KMeans.swift [hunk 1]
```diff
@@ -71,7 +71,12 @@ func kmeans<T : ClusteredType>(
         previousError = error
     } while abs(error - previousError) > threshold
     
-    return map(Zip2(centroids, clusterSizes)) { Cluster(centroid: $0, size: $1) }
+        var clusters = [Cluster<T>]()
+        for (first, second) in Zip2Sequence(centroids, clusterSizes) {
```
Comment (indragiek) [inline DominantColor/Shared/KMeans.swift]: Is there a reason why `Zip2Sequence.map` can't be used here instead of the for loop?

**Comment by indragiek** [comment]
Looks good, just one comment to address. Thank you!

**Comment by rshevchuk** [comment]
Thanks for the catch :) I found no reason there for the map replacement. Map can be used here. I fixed this and also added one fix for Swift 3 which is related to this https://github.com/apple/swift-evolution/blob/master/proposals/0004-remove-pre-post-inc-decrement.md

**Comment by indragiek** [comment]
👍 Thank you very much!

**Pull request merged**